routers/repo/view.go

```diff
@@ -264,14 +264,15 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 // Home render repository home page
 func Home(ctx *context.Context) {
 	if len(ctx.Repo.Repository.Units) > 0 {
-		tp := ctx.Repo.Repository.Units[0].Type
-		if tp == models.UnitTypeCode {
-			renderCode(ctx)
-			return
+		for _, unit := range ctx.Repo.Repository.Units {
+			if unit.Type == models.UnitTypeCode {
+				renderCode(ctx)
+				return
+			}
 		}
 
-		unit, ok := models.Units[tp]
-		if ok {
+		unit, ok := models.Units[ctx.Repo.Repository.Units[0].Type]
+		if ok && unit.URI != models.UnitCode.URI {
```

Comment:
Seems it will never be models.UnitCode.URI

Reply:
It shouldn't, but to be sure we don't have dead loop.